### PR TITLE
Fix doc-comments types for fields of inline objects

### PR DESCRIPTION
**Pull request by lukyanovas**

### CODE REVIEW

**Review by mrjohn0011 — DISMISSED** [review]

File: js/data/custom_store.js [hunk 1]
```diff
@@ -260,7 +260,7 @@ var CustomStore = Store.inherit({
          * @type_function_param1_field8 requireTotalCount:boolean
          * @type_function_param1_field9 searchValue:object
          * @type_function_param1_field10 searchOperation:string
-         * @type_function_param1_field11 searchExpr:getter|array
+         * @type_function_param1_field11 searchExpr:getter|Array<string>
```
Comment (San4es) [inline js/data/custom_store.js]: IIRC, the correct value is: 
`getter|Array<getter>`

/cc @AlekseyMartynov

File: js/data/data_source/data_source.js [hunk 2]
```diff
@@ -190,7 +190,7 @@ var DataSource = Class.inherit({
         /**
         * @name DataSourceOptions_store
         * @publicName store
-        * @type Store|Array<any>|Object
+        * @type Store|StoreOptions|Array<any>|Object|any
```
Comment (San4es) [inline js/data/data_source/data_source.js]: What does `Object` value mean here? I suspect that `Object` is `StoreOptions` here.

/cc @AlekseyMartynov

File: js/data/custom_store.js (hunk 1 above)
Comment (AlekseyMartynov) [reply js/data/custom_store.js]: That's right. See https://github.com/DevExpress/DevExtreme/blob/9cb3052fa31bca1021805b79a16e08e37a76e4bc/js/data/data_source/data_source.js#L891-L893

File: js/data/data_source/data_source.js (hunk 2 above)
Comment (AlekseyMartynov) [reply js/data/data_source/data_source.js]: https://github.com/DevExpress/DevExtreme/blob/9cb3052fa31bca1021805b79a16e08e37a76e4bc/js/data/data_source/data_source.js#L116-L126

`store` can be:
- array
- store options
- a plain object with the `load` member

I suggest that we keep `any` and remove `Object`

File: js/viz/docs/docvectormap.js [hunk 3]
```diff
@@ -1393,8 +1393,8 @@ var dxVectorMap = {
     * @type_function_param1 e:object
     * @type_function_param1_field1 component:object
     * @type_function_param1_field2 element:object
-    * @type_function_param1_field3 jQueryEvent:jQuery-event object
-    * @type_function_param1_field4 target:Marker object
+    * @type_function_param1_field3 jQueryEvent:jQueryEvent
+    * @type_function_param1_field4 target:object
```
Comment (roman-simionov) [inline js/viz/docs/docvectormap.js]: markerObjects

**Review by roman-simionov — CHANGES_REQUESTED** [review]
see my comment in code

File: js/data/data_source/data_source.js (hunk 2 above)
Comment (lukyanovas) [reply js/data/data_source/data_source.js]: @AlekseyMartynov Can the store be the instance of Store object? If not we should also remove 'Store' type from DataSourceOptions_store

Comment (lukyanovas) [reply js/data/data_source/data_source.js]: yes, I see https://github.com/DevExpress/DevExtreme/blob/9cb3052fa31bca1021805b79a16e08e37a76e4bc/js/data/data_source/data_source.js#L106

File: js/ui/pivot_grid/ui.pivot_grid.js [hunk 4]
```diff
@@ -304,17 +304,17 @@ var PivotGrid = Widget.inherit({
...
             * @type_function_param1_field9 rowIndex:number
-            * @type_function_param1_field10 dataFields:array
-            * @type_function_param1_field11 rowFields:array
-            * @type_function_param1_field12 columnFields:array
-            * @type_function_param1_field13 field:object
+            * @type_function_param1_field10 dataFields:Array<Object>
+            * @type_function_param1_field11 rowFields:Array<Object>
+            * @type_function_param1_field12 columnFields:Array<Object>
+            * @type_function_param1_field13 field:Object
             * @extends Action
```
Comment (roman-simionov) [inline js/ui/pivot_grid/ui.pivot_grid.js]: @type_function_param1_field10 dataFields:Array<PivotGridDataSourceOptions_fields>
@type_function_param1_field11 rowFields:Array<PivotGridDataSourceOptions_fields>
 @type_function_param1_field12 columnFields:Array<PivotGridDataSourceOptions_fields>
 @type_function_param1_field13 PivotGridDataSourceOptions_fields

File: js/ui/pivot_grid/ui.pivot_grid.js [hunk 5]
```diff
@@ -677,15 +677,15 @@ var PivotGrid = Widget.inherit({
...
+            * @type_function_param1_field6 cell:dxPivotGridPivotGridCell
             * @type_function_param1_field7 rowIndex:number
             * @type_function_param1_field8 columnIndex:number
-            * @type_function_param1_field9 columnFields:Array
-            * @type_function_param1_field10 rowFields:Array
-            * @type_function_param1_field11 dataFields:Array
+            * @type_function_param1_field9 columnFields:Array<Object>
+            * @type_function_param1_field10 rowFields:Array<Object>
+            * @type_function_param1_field11 dataFields:Array<Object>
             * @type_function_param1_field12 jQueryEvent:jQueryEvent
```
Comment (roman-simionov) [inline js/ui/pivot_grid/ui.pivot_grid.js]: @type_function_param1_field9 columnFields:Array<PivotGridDataSourceOptions_fields>
 @type_function_param1_field10 rowFields:Array<PivotGridDataSourceOptions_fields>
 @type_function_param1_field11 dataFields:Array<PivotGridDataSourceOptions_fields>

File: js/data/data_source/data_source.js [hunk 6]
```diff
@@ -190,7 +190,7 @@ var DataSource = Class.inherit({
         /**
         * @name DataSourceOptions_store
         * @publicName store
-        * @type Store|Array<any>|Object
+        * @type Store|StoreOptions|Array<any>|Object|string|any
```
Comment (AlekseyMartynov) [inline js/data/data_source/data_source.js]: What's the case for `string`?

Comment (lukyanovas) [reply js/data/data_source/data_source.js]: What about creating a store from URL?

File: js/ui/pivot_grid/ui.pivot_grid.field_chooser.js [hunk 7]
```diff
@@ -131,7 +131,7 @@ var FieldChooser = BaseFieldChooser.inherit({
             * @publicName onContextMenuPreparing
             * @type function(e)
             * @type_function_param1 e:object
-            * @type_function_param1_field4 items:array
+            * @type_function_param1_field4 items:Array<Object>
             * @type_function_param1_field5 area:string
             * @type_function_param1_field6 field:object
```
Comment (roman-simionov) [inline js/ui/pivot_grid/ui.pivot_grid.field_chooser.js]: @type_function_param1_field6 field: PivotGridDataSourceOptions_fields

File: js/data/data_source/data_source.js (hunk 6 above)
Comment (AlekseyMartynov) [reply js/data/data_source/data_source.js]: Data Source can be [specified as an URL](https://js.devexpress.com/Documentation/Guide/Data_Layer/Data_Layer/#Creating_DataSource/From_url), Store [cannot](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/).

**Review by roman-simionov — APPROVED** [review]
pivot grid and charts look good

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**